guides/keras_nlp/fnet_classification.py

```diff
@@ -0,0 +1,380 @@
+"""
+Title: Text Classification using FNet
+Author: [Abheesht Sharma](https://github.com/abheesht17/)
+Date created: 2021/06/01
+Last modified: 2021/06/01
+Description: Text Classification on the SST-2 Dataset using KerasNLP's `FNetEncoder` layer
+"""
+
+"""
+## Introduction
+
+In this example, we will demonstrate the ability of FNet to achieve comparable
+results with a vanilla Transformer model on the text classification task.
+We will be using the SST-2 dataset (belongs to the GLUE benchmark), which is a
+collection of movie reviews labelled either positive or negative (sentiment
+analysis).
+
+To build the tokenizer, model, etc., we will use components from
+[KerasNLP](https://github.com/keras-team/keras-nlp). KerasNLP makes life easier
+for people who want to build NLP pipelines! :)
+
+### Model
+
+In 2017, a paper titled [Attention is All You Need](https://arxiv.org/abs/1706.03762)
```

Comment:
I would generally tighten up this section. Examples shouldn't have a ton of offhand comments, we should focus on what is show in this guide. This reads a little too much like a blog currently.

Roughly, we should just say...

 - BERT, RoBERTa, etc have shown the effectiveness of using transformers to compute a rich embedding for input text.
 - However transformers are expensive, an ongoing question is how to lower the compute requirements.
 - In this guide we will focus on FNet, which replace the expensive attention mechanism with a Fourier transform.
 - We will show how this can speed up training, without significantly degrading performance.